modules/multisignatures.js

```diff
@@ -80,158 +80,221 @@ function Multisignatures(cb, scope) {
+					{ signature, members, transaction }
+				);
+				return isValid;
 			}
 
-			modules.accounts.getAccount(
-				{
-					address: multisignatureTransaction.senderId,
-				},
-				(err, sender) => {
-					if (err) {
-						return setImmediate(cb, err);
-					} else if (!sender) {
-						return setImmediate(cb, 'Sender not found');
-					}
-					multisignatureTransaction.signatures =
-						multisignatureTransaction.signatures || [];
-					multisignatureTransaction.signatures.push(transaction.signature);
-					multisignatureTransaction.ready = Multisignature.prototype.ready(
-						multisignatureTransaction,
-						sender
-					);
-
-					library.bus.message('signature', transaction, true);
-					return setImmediate(cb);
-				}
+			// Try to verify the signature
+			isValid = library.logic.transaction.verifySignature(
+				transaction,
+				signature.publicKey,
+				signature.signature
+			);
+		} else {
+			// publicKey is not there - we need to iterate through all members of multisignature account in transaction
+			isValid = members.find(member =>
+				// Try to verify the signature
+				library.logic.transaction.verifySignature(
+					transaction,
+					member,
+					signature.signature
+				)
 			);
-		}, cb);
+		}
+	} catch (e) {
+		library.logger.error('Unable to process signature, verification failed.', {
+			signature,
+			members,
+			transaction,
+			error: e.stack,
+		});
 	}
+	return isValid;
+};
 
-	if (!multisignatureTransaction) {
-		return setImmediate(cb, 'Transaction not found');
+/**
+ * Process signature for multisignature account creation, transaction type 4 (MULTI)
+ *
+ * @private
+ * @param {Object} signature - Signature data
+ * @param {string} [signature.publicKey] - Public key of account that created the signature (optional)
+ * @param {string} signature.transactionId - Id of transaction that signature was created for
+ * @param {string} signature.signature - Actual signature
+ * @param {Object} transaction - Corresponding transaction grabbed from transaction pool
+ * @param {function} cb - Callback function
+ * @returns {setImmediateCallback} cb, err
+ */
+__private.processSignatureForMultisignatureAccountCreation = (
+	signature,
+	transaction,
+	cb
+) => {
+	// Normalize members of multisignature account from transaction
+	const members = transaction.asset.multisignature.keysgroup.map(member =>
+		member.substring(1) // Remove first character, which is '+'
+	);
+
+	// Check if signature is valid
+	if (!__private.isValidSignature(signature, members, transaction)) {
+		return setImmediate(cb, 'Unable to process signature, verification failed');
 	}
 
-	if (multisignatureTransaction.type === transactionTypes.MULTI) {
-		multisignatureTransaction.signatures =
-			multisignatureTransaction.signatures || [];
+	// Add signature to transaction
+	transaction.signatures.push(signature.signature);
+	// Check if transaction is ready to be processed
+	transaction.ready = Multisignature.prototype.ready(
```

Comment:
Please add `scope.logic.multisignature` to `library` and use its object instead of directly accessing a method through `prototype`. Calling a prototype method directly is only used while implementing object specific behavior to override something in object.